### PR TITLE
🐛Source Jira: add sprint information from team-managed project

### LINKED ISSUE #8281: Source Jira: adds sprint information from team-managed project
## Tell us about the problem you're trying to solve

* Deployment: Docker
* Airbyte Version: 0.35.5-alpha
* Source name/version: Jira 0.2.15
* Destination name/version: BigQuery 0.5.0

Using the Jira connector, I want to extract informations coming from both company-managed and team-managed projects (doc [here](https://support.atlassian.com/jira-software-cloud/docs/what-are-team-managed-and-company-managed-projects/)). I am able to extract bassic information like the `issues` stream, but the `sprints` and `sprint_issues` streams are empty even if there is sprint on corresponding team-managed projects. It work well with company-managed projects.

## Describe the solution you’d like
I want to see the same level of detail for team-managed projects than we have on company-managed projects.

## Describe the alternative you’ve considered or used
I am thinking of migrating my Jira project to company-managed.

## Additional context

A few lines of logs:
```
source - 2021-11-23 02:14:09 INFO () DefaultAirbyteStreamFactory(internalLog):97 - Starting syncing SourceJira
source - 2021-11-23 02:14:09 INFO () DefaultAirbyteStreamFactory(internalLog):97 - Syncing stream: issues 
source - 2021-11-23 02:14:11 INFO () DefaultAirbyteStreamFactory(internalLog):97 - Read 15 records from issues stream
source - 2021-11-23 02:14:11 INFO () DefaultAirbyteStreamFactory(internalLog):97 - Syncing stream: projects 
source - 2021-11-23 02:14:11 INFO () DefaultAirbyteStreamFactory(internalLog):97 - Read 1 records from projects stream
source - 2021-11-23 02:14:11 INFO () DefaultAirbyteStreamFactory(internalLog):97 - Syncing stream: sprint_issues 
destination - 2021-11-23 02:14:11 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-23 02:14:11 INFO i.a.i.d.b.BigQueryDestination(main):334 - {} - starting destination: class io.airbyte.integrations.destination.bigquery.BigQueryDestination
destination - 2021-11-23 02:14:11 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-23 02:14:11 INFO i.a.i.b.IntegrationRunner(run):76 - {} - Running integration: io.airbyte.integrations.destination.bigquery.BigQueryDestination
destination - 2021-11-23 02:14:11 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-23 02:14:11 INFO i.a.i.b.IntegrationCliParser(parseOptions):115 - {} - integration args: {catalog=destination_catalog.json, write=null, config=destination_config.json}
destination - 2021-11-23 02:14:11 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-23 02:14:11 INFO i.a.i.b.IntegrationRunner(run):80 - {} - Command: WRITE
destination - 2021-11-23 02:14:11 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-23 02:14:11 INFO i.a.i.b.IntegrationRunner(run):81 - {} - Integration config: IntegrationConfig{command=WRITE, configPath='destination_config.json', catalogPath='destination_catalog.json', statePath='null'}
source - 2021-11-23 02:14:12 INFO () DefaultAirbyteStreamFactory(internalLog):97 - Read 0 records from sprint_issues stream
source - 2021-11-23 02:14:12 INFO () DefaultAirbyteStreamFactory(internalLog):97 - Syncing stream: sprints 
destination - 2021-11-23 02:14:12 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-23 02:14:12 WARN c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword examples - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
destination - 2021-11-23 02:14:12 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-23 02:14:12 WARN c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword airbyte_secret - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
destination - 2021-11-23 02:14:12 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-23 02:14:12 INFO i.a.i.d.b.BigQueryDestination(getLoadingMethod):309 - {} - Selected loading method is set to: STANDARD
```

But I have some sprints on the project:
image

## Are you willing to submit a PR?

I can dig a bit to understand how to work on that but I will need some guidance!

I have already discussed of this on [this](https://app.slack.com/client/T01AB4DDR2N/slack-connect/thread/C01MFR03D5W-1637809139.433900) slack thread.

Thanks !

### PR DESCRIPTION
**Pull request by darynaishchenko**
In `Sprints` stream we had `board["type"] == "scrum"` check. It was the problem of loosing data from team-managed projects, which have simple board type.
Solution: adding `board["type"] in ["scrum", "simple"]` to validate both types that can contain sprints in it.

### CODE REVIEW

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py [hunk 1]
```diff
@@ -1016,7 +1016,15 @@ def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwarg
...
-        yield from super().read_records(stream_slice={"screen_id": screen_id}, **kwargs)
+        for screen_tab in super().read_records(stream_slice={"screen_id": screen_id}, **kwargs):
+            """
+            For some projects jira creates screens automatically, which does not present in UI, but exist in screens stream.
+            We receive 400 error "Screen with id {screen_id} does not exist" for tabs by these screens.
+            """
+            if screen_tab.get("errorMessages") and re.match(r"Screen with id \d* does not exist", screen_tab["errorMessages"][0]):
+                self.logger.info("Could not get screen tab for %s screen id. Reason: %s", screen_id, screen_tab["errorMessages"][0])
+            else:
+                yield screen_tab
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-jira/source_jira/streams.py:1027]:
```suggestion
            bad_request_reached = re.match(r"Screen with id \d* does not exist", screen_tab.get("errorMessages", [''])[0])
            if bad_request_reached:
                self.logger.info("Could not get screen tab for %s screen id. Reason: %s", screen_id, screen_tab["errorMessages"][0])
                return
            yield screen_tab
```

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py [hunk 2]
```diff
@@ -1072,7 +1080,7 @@ def path(self, stream_slice: Mapping[str, Any], **kwargs) -> str:
 
     def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
         for board in read_full_refresh(self.boards_stream):
-            if board["type"] == "scrum":
+            if board["type"] in ["scrum", "simple"]:
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-jira/source_jira/streams.py:1083]:
```suggestion
            available_board_types = ["scrum", "simple"]
            if board["type"] in available_board_types:
```

File: airbyte-integrations/connectors/source-jira/Dockerfile [hunk 3]
```diff
@@ -12,5 +12,5 @@ RUN pip install .
 ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
-LABEL io.airbyte.version=0.3.7
+LABEL io.airbyte.version=0.3.8
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-jira/Dockerfile:15]: The version in the` metadata.yaml` file needs to be updated as well.

**Review by lazebnyi — CHANGES_REQUESTED** [review]
Looks good, but there are some minor changes that need to be made. Once those changes are made, I will approve it.

File: airbyte-integrations/connectors/source-jira/Dockerfile (hunk 3 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-jira/Dockerfile:15]: done https://github.com/airbytehq/airbyte/pull/25798/files#diff-60f74e883cc95c68ef66f89e9e8e72f610300f153fde36acc1d70a1ec73d41ffR8

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**